sat_vapor_pres/include/sat_vapor_pres_k.inc

```diff
@@ -326,86 +176,98 @@
 ! differencing des values in the table
 
      do i = 2, table_size-1
-       D2TABLE3(i) = 0.25*dtinvl*(DTABLE3(i+1)-DTABLE3(i-1))
+       D2TABLE3(i) = 0.25_r8_kind*dtinvl*(DTABLE3(i+1)-DTABLE3(i-1))
      enddo
 ! one-sided derivatives at boundaries
 
-     D2TABLE3(1) = 0.50*dtinvl*(DTABLE3(2)-DTABLE3(1))
+     D2TABLE3(1) = 0.50_r8_kind*dtinvl*(DTABLE3(2)-DTABLE3(1))
 
-     D2TABLE3(table_size) = 0.50*dtinvl*&
-          (DTABLE3(table_size)-DTABLE3(table_size-1))
+     D2TABLE3(table_size) = 0.50_r8_kind*dtinvl*(DTABLE3(table_size)-DTABLE3(table_size-1))
    endif
 
       use_exact_qs = use_exact_qs_input
       module_is_initialized = .true.
 
- end subroutine sat_vapor_pres_init_k
+  end subroutine SAT_VAPOR_PRES_INIT_K_
 
 !#######################################################################
 
- function compute_es_k(tem, TFREEZE) result (es)
- real, intent(in) :: tem(:), TFREEZE
- real :: es(size(tem,1))
+ function COMPUTE_ES_K_(tem, TFREEZE) result (es)
+ real(kind=FMS_SVP_KIND_), intent(in) :: tem(:), TFREEZE
+ real(kind=FMS_SVP_KIND_) :: es(size(tem,1))
 
- real    :: x, esice, esh2o, TBASW, TBASI
+ real(kind=FMS_SVP_KIND_)  :: x, esice, esh2o, TBASW, TBASI
  integer :: i
- real, parameter :: ESBASW = 101324.60
- real, parameter :: ESBASI =    610.71
 
-   TBASW = TFREEZE+100.
+ integer, parameter :: kindl=FMS_SVP_KIND_ !< local kind parameter
+
+ real(kind=FMS_SVP_KIND_), parameter :: ESBASW = real(101324.60d0,FMS_SVP_KIND_)
+ real(kind=FMS_SVP_KIND_), parameter :: ESBASI = real(610.71d0,FMS_SVP_KIND_)
+
+ real(FMS_SVP_KIND_), parameter :: one=real(1.0d0,FMS_SVP_KIND_)
+ real(FMS_SVP_KIND_), parameter :: ten=real(10.0d0,FMS_SVP_KIND_)
+
+   TBASW = TFREEZE+100.0_kindl
    TBASI = TFREEZE
 
    do i = 1, size(tem)
 
 !  compute es over ice
 
      if (tem(i) < TBASI) then
-         x = -9.09718*(TBASI/tem(i)-1.0) - 3.56654*log10(TBASI/tem(i)) &
-             +0.876793*(1.0-tem(i)/TBASI) + log10(ESBASI)
-         esice =10.**(x)
+         x = -9.09718_kindl*(TBASI/tem(i)-one)   &
+             -3.56654_kindl*log10(TBASI/tem(i)) &
+             +0.876793_kindl*(one-tem(i)/TBASI) + log10(ESBASI)
+         esice =ten**(x)
      else
-         esice = 0.
+         esice = 0.0_kindl
      endif
 
 !  compute es over water greater than -20 c.
 !  values over 100 c may not be valid
 !  see smithsonian meteorological tables page 350.
 
-     if (tem(i) > -20.+TBASI) then
-         x = -7.90298*(TBASW/tem(i)-1.0) + 5.02808*log10(TBASW/tem(i)) &
-             -1.3816e-07*(10.0**((1.0-tem(i)/TBASW)*11.344)-1.0)        &
-             +8.1328e-03*(10.0**((TBASW/tem(i)-1.0)*(-3.49149))-1.0)    &
+     if (tem(i) > -20.0_kindl+TBASI) then
+         x = -7.90298_kindl*(TBASW/tem(i)-one)   &
+             +5.02808_kindl*log10(TBASW/tem(i)) &
+             -real(1.3816d-07,FMS_SVP_KIND_)*(ten**((one-tem(i)/TBASW)*11.344_kindl)-one) &
```

Comment:
Would this still work if the `d` gives you a quad precision? gcc would make it double the default so I'm wondering if casting from that would give you a different result than the original code.

Reply:
@rem1776  you are right.  kind(1.d0) with gcc is 16... oh dear.

Reply:
or in the case of 1.e10, compile with default 64